Adaptors/Amqp/src/PullQueueStorage.cs

```diff
@@ -46,66 +45,64 @@ public class PullQueueStorage : QueueStorage, IPullQueueStorage
-                                                                                      $"{options.PartitionId}###SenderLink{i}",
-                                                                                      $"{options.PartitionId}###q{i}")))
-                         .ToArray();
-
     logger_ = logger;
   }
 
-  public new Task<HealthCheckResult> Check(HealthCheckTag tag)
-    => Task.FromResult(isInitialized_
-                         ? HealthCheckResult.Healthy()
-                         : HealthCheckResult.Unhealthy());
-
   public override async Task Init(CancellationToken cancellationToken)
   {
-    if (!isInitialized_)
+    if (!IsInitialized)
     {
+      await ConnectionAmqp.Init(cancellationToken)
+                          .ConfigureAwait(false);
+
+      var session = new Session(ConnectionAmqp.Connection);
+
+      receivers_ = Enumerable.Range(0,
+                                    NbLinks)
+                             .Select(i => new AsyncLazy<IReceiverLink>(() =>
+                                                                       {
+                                                                         var rl = new ReceiverLink(session,
+                                                                                                   $"{Options.PartitionId}###ReceiverLink{i}",
+                                                                                                   $"{Options.PartitionId}###q{i}");
+
+                                                                         /* linkCredit_: the maximum number of messages the
+                                                                       * remote peer can send to the receiver.
+                                                                       * With the goal of minimizing/deactivating
+                                                                       * prefetching, a value of 1 gave us the desired
+                                                                       * behavior. We pick a default value of 2 to have "some cache". */
+                                                                         rl.SetCredit(Options.LinkCredit);
+                                                                         return rl;
+                                                                       }))
+                             .ToArray();
+
+      senders_ = Enumerable.Range(0,
+                                  NbLinks)
+                           .Select(i => new AsyncLazy<ISenderLink>(() => new SenderLink(session,
+                                                                                        $"{Options.PartitionId}###SenderLink{i}",
+                                                                                        $"{Options.PartitionId}###q{i}")))
+                           .ToArray();
+
       var senders   = Task.WhenAll(senders_.Select(async lazy => await lazy));
```

Comment:
Couldn't you just do `Task.WhenAll(senders_)` ?

Reply:
No there is the alzy async thingy in it

Reply:
I thought `AsyncLazy` was awaitable.